app/scripts/controllers/swaps.js

```diff
@@ -291,6 +297,24 @@ export default class SwapsController {
       destinationTokenInfo: fetchParamsMetaData.destinationTokenInfo,
     }));
 
+    if (chainId === CHAIN_IDS.OPTIMISM && Object.values(newQuotes).length > 0) {
+      await Promise.all(
+        Object.values(newQuotes).map(async (quote) => {
+          if (quote.trade) {
+            const multiLayerL1TradeFeeTotal = await fetchEstimatedL1Fee(
```

Comment:
As of [this PR](https://github.com/MetaMask/metamask-extension/pull/16891) (merged earlier today) this function no longer takes a provider as its first arg

Reply:
Yea so it seems like since this is called in the context of the background it doesn't have access to that global.

Reply:
here's a diff for change that I suggest we make:
[fix-eth-provider.txt](https://github.com/MetaMask/metamask-extension/files/10279362/fix-eth-provider.txt)

Reply:
```
diff --git a/app/scripts/controllers/swaps.js b/app/scripts/controllers/swaps.js
index 8b5a1c331..ea7d45e4c 100644
--- a/app/scripts/controllers/swaps.js
+++ b/app/scripts/controllers/swaps.js
@@ -301,10 +301,13 @@ export default class SwapsController {
       await Promise.all(
         Object.values(newQuotes).map(async (quote) => {
           if (quote.trade) {
-            const multiLayerL1TradeFeeTotal = await fetchEstimatedL1Fee({
-              txParams: quote.trade,
-              chainId,
-            });
+            const multiLayerL1TradeFeeTotal = await fetchEstimatedL1Fee(
+              {
+                txParams: quote.trade,
+                chainId,
+              },
+              this.ethersProvider,
+            );
             quote.multiLayerL1TradeFeeTotal = multiLayerL1TradeFeeTotal;
           }
           return quote;
diff --git a/ui/helpers/utils/optimism/fetchEstimatedL1Fee.js b/ui/helpers/utils/optimism/fetchEstimatedL1Fee.js
index da9477fd2..95c073665 100644
--- a/ui/helpers/utils/optimism/fetchEstimatedL1Fee.js
+++ b/ui/helpers/utils/optimism/fetchEstimatedL1Fee.js
@@ -19,8 +19,11 @@ const OPTIMISM_GAS_PRICE_ORACLE_ABI = [
 const OPTIMISM_GAS_PRICE_ORACLE_ADDRESS =
   '0x420000000000000000000000000000000000000F';
 
-export default async function fetchEstimatedL1Fee(txMeta) {
-  const provider = new Web3Provider(global.ethereumProvider, 10);
+export default async function fetchEstimatedL1Fee(txMeta, ethersProvider) {
+  const provider = global.ethereumProvider
+    ? new Web3Provider(global.ethereumProvider, 10)
+    : ethersProvider;
+
   if (process.env.IN_TEST) {
     provider.detectNetwork = async () => ({
       name: 'optimism',
```